bundles/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/internal/handler/ZonePlayerHandler.java

```diff
@@ -100,96 +98,63 @@
      */
-    private static final int DEFAULT_REFRESH_INTERVAL = 60;
-
-    private final Map<String, String> stateMap = Collections.synchronizedMap(new HashMap<>());
-
-    private List<SonosMusicService> musicServices;
-
+    private final Object notificationLock = new Object();
     private final Object upnpLock = new Object();
-
     private final Object stateLock = new Object();
+    private final Object jobLock = new Object();
 
-    private final SonosStateDescriptionOptionProvider stateDescriptionProvider;
-
-    private final Runnable pollingRunnable = () -> {
-        try {
-            logger.debug("Polling job");
+    private final Map<String, String> stateMap = Collections.synchronizedMap(new HashMap<>());
 
-            // First check if the Sonos zone is set in the UPnP service registry
-            // If not, set the thing state to OFFLINE and wait for the next poll
-            if (!isUpnpDeviceRegistered()) {
-                logger.debug("UPnP device {} not yet registered", getUDN());
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                        "@text/offline.upnp-device-not-registered [\"" + getUDN() + "\"]");
-                synchronized (upnpLock) {
-                    subscriptionState = new HashMap<>();
-                }
-                return;
-            }
+    private ScheduledFuture<?> pollingJob;
+    private SonosZonePlayerState savedState = null;
 
-            // Check if the Sonos zone can be joined
-            // If not, set the thing state to OFFLINE and do nothing else
-            updatePlayerState();
-            if (getThing().getStatus() != ThingStatus.ONLINE) {
-                return;
-            }
+    private Map<String, Boolean> subscriptionState = new HashMap<>();
 
-            addSubscription();
+    /**
+     * configurable notification timeout (in seconds)
+     */
+    private Integer notificationTimeout = null;
 
-            updateZoneInfo();
-            updateLed();
-            updateSleepTimerDuration();
-        } catch (Exception e) {
-            logger.debug("Exception during poll: {}", e.getMessage(), e);
-        }
-    };
+    /**
+     * {@link ThingHandler} instance of the coordinator speaker used for control delegation
+     */
+    private ZonePlayerHandler coordinatorHandler;
 
-    private final String opmlUrl;
+    private List<SonosMusicService> musicServices;
 
     public ZonePlayerHandler(ThingRegistry thingRegistry, Thing thing, UpnpIOService upnpIOService, String opmlUrl,
             SonosStateDescriptionOptionProvider stateDescriptionProvider) {
         super(thing);
         this.localThingRegistry = thingRegistry;
         this.opmlUrl = opmlUrl;
         logger.debug("Creating a ZonePlayerHandler for thing '{}'", getThing().getUID());
-        if (upnpIOService != null) {
-            this.service = upnpIOService;
-        }
+        this.service = upnpIOService;
```

Comment:
I agree that the null-check like it was before does nothing,  but I wonder if removing is the correct way. Are you sure that every access to `service` is null-safe? Otherwise it would be probably better to throw an `IllegalArgumentException`. 

The correct way would obviously be to add proper null-annotations, but that is out-of-scope of this PR.

Reply:
The Sonos handler factory bundle will not start without this bundle. It is set as a reference:
https://github.com/openhab/openhab-addons/blob/2.5.x/bundles/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/internal/SonosHandlerFactory.java#L74
So it cannot be null.
That is exactly the same thing as for thingRegistry or stateDescriptionProvider.